syntax/lexer.go

```diff
@@ -1193,6 +1193,11 @@ func Quote(s string) (_ string, ok bool) {
 			return "", false
 		}
 		if r == utf8.RuneError || !unicode.IsPrint(r) {
+			if lang == LangPOSIX {
+				// POSIX shell lacks $'.
+				// Without it, we can't use escape sequences such as \x.
+				return "", false
```

Comment:
Wouldn't it be useful to return an error instead of a boolean? Then if this failed, people would have a message explaining why.

The same applies to other possible errors.

Reply:
What would you do with an error value?

Having some information like the offset might be useful, but beyond that I'm not sure.

Reply:
It's more about helping the user to understand what gone wrong.

"syntax: it is not possible to quote \x in POSIX mode" is more useful than "false" when trying to understand what happened.

Just a suggestion, though. Likely not a big deal to keep as is if you want. 🙂

Reply:
Yeah that's fair. The signature is from when we only returned false on null bytes, so back then it made less sense to use an error. I'll add one.

Reply:
Done.